src/components/StaffList/index.jsx

```diff
@@ -33,17 +81,83 @@ class StaffList extends React.Component {
     }, () => this.props.input.onChange(this.state.staffList));
   }
 
+  /**
+   * The callback method for when a user selects a suggestion in the autocomplete list.
+   * Responsible for updating the search input text after selection and for updating the redux
+   * state of selected options.
+   *
+   * @param selectedValue - the selected autocomplete option
+   * @returns {string} - the value for the search input box after a selection
+   */
+  getSuggestionValue(selectedValue) {
+    const { value } = this.props.input;
+    const newInstructor = (selectedValue.id === 'new');
+    const instructorAlreadySelected =
+      (value.filter(staff => staff.uuid === selectedValue.text.uuid)).length > 0;
```

Comment:
I think you want to be altering the state through `staffList` and then pushing those changes to `redux-form` like the older change functions do.

Also, I think you can do this check more intuitively with `.some`

Something like:
`staffList.some(staffer => staffer.uuid === selectedValue.text.uuid)`

Reply:
👍